lib/ansible/modules/network/checkpoint/cp_gaia_physical_interface.py

```diff
@@ -0,0 +1,158 @@
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+from __future__ import (absolute_import, division, print_function)
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = """
+author: Yuval Feiger (@chkp-yuvalfe)
+description:
+- Set Physical interface
+module: cp_gaia_physical_interface
+options:
+  auto_negotiation:
+    description: Activating Auto-Negotiation will skip the speed and duplex configuration
+    required: false
+    type: bool
+  comments:
+    description: interface Comments.
+    required: false
+    type: str
+  duplex:
+    description: Duplex for the interface. Duplex is not relevant when 'auto-negotiation' is enabled
+    required: false
+    type: str
+  enabled:
+    description: interface State.
+    required: false
+    type: bool
+  ipv4_address:
+    description: Interface IPv4 address.
+    required: false
+    type: str
+  ipv4_mask_length:
+    description: Interface IPv4 address mask length.
+    required: false
+    type: str
+  ipv6_address:
+    description: Interface IPv6 address.
+    required: false
+    type: str
+  ipv6_autoconfig:
+    description: Configure IPv6 auto-configuration.
+    required: false
+    type: bool
+  ipv6_mask_length:
+    description: Interface IPv6 address mask length.
+    required: false
+    type: str
+  mac_addr:
+    description: Configure hardware address.
+    required: false
+    type: str
+  monitor_mode:
+    description: set monitor mode for the interface off/on.
+    required: false
+    type: bool
+  mtu:
+    description: interface mtu.
+    required: false
+    type: int
+  name:
+    description: interface name.
+    required: true
+    type: str
+  rx_ringsize:
+    description: Set receive buffer size for interface.
+    required: false
+    type: int
+  speed:
+    description: Interface link speed. Speed is not relevant when 'auto-negotiation' is enabled
+    required: false
+    type: str
+  tx_ringsize:
+    description: Set transmit buffer size for interfaces.
+    required: false
+    type: int
+short_description: Set Physical interface
+version_added: '2.9'
+
+"""
+
+EXAMPLES = """
+- name: Set comment field of a physical interface
+  cp_gaia_physical_interface:
+    comments: eth0 interface
+    enabled: true
+    name: eth0
```

Comment:
`state` option is needed by Ansible module to make it aligned to Ansible conventio std., and once `state` option is included plz include the example for `state: absent` i.e. removal example as well (ref: https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/network/checkpoint/checkpoint_host.py#L47)

Reply:
`state` parameter won't be useful here - since it physical interface, you cannot add or remove it, just edit the configurations of existing interface.
I can add `state` parameter but I think it will just confuse users because I don't have something to do with the `absent`/`present` values.

Reply:
@chkp-yuvalfe even if the configuration is edited it would be either added/removed correct?